tests/test_GitRepository.cc

```diff
@@ -0,0 +1,290 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+
+#include <git2.h>
+#include <gul14/catch.h>
+#include "taskolib/GitRepository.h"
+#include "taskolib/serialize_sequence.h"
+#include "taskolib/Step.h"
+#include "taskolib/Sequence.h"
+#include <filesystem>
+#include <fstream>
+
+using namespace task;
+
+
+/**
+ *  Creates a directory and store files in it.
+ * Filestructure: \n
+ * sequences/     \n
+ *      $name$/   \n
+ *          file0.txt   << $msg$ /n file0  \n
+ *          file1.txt   << $msg$ /n file1  \n
+ *          ...
+ * \param name name of the directory
+ * \param nr_files number of files to be created
+ * \param msg What to write to the file
+ */
+void create_testfiles(const std::filesystem::path& name, size_t nr_files, const std::string& msg)
+{
+    std::filesystem::path p = "sequences" / name;
+    std::filesystem::create_directories(p);
+    for (size_t i = 0; i < nr_files; i++)
+    {
+        // msg
+        // file i
+        std::ofstream f(p/gul14::cat("file", i, ".txt"));
+        f << msg << gul14::cat("\nfile", i);
+    }  
+}
+
+
+TEST_CASE("GitRepository Wrapper Test all", "[GitWrapper]")
+{
+    /**
+     * Create files in a directory and then initialize the git repository within.
+     * The initial commit must be empty as no staging has been done.
+     * Check for the initialization of member variables
+     * Check if a repository is created (HEAD exists)
+     * Check if a comit was created (initial commit)
+     * 
+     */
+    SECTION("Construct GitRepository object")
+    {
+        create_testfiles("unit_test_1", 2, "Construct");
+
+        // Create Git Library
+        GitRepository gl{"sequences"};
+
+        REQUIRE(not gl.get_path().empty());
+        REQUIRE(gl.get_path() == "sequences");
+        git_reference *temp;
+        REQUIRE(git_repository_head(&temp, gl.get_repo()) == 0);
+        
+        
+        // Test if repo_ got initialized
+        REQUIRE(gl.get_last_commit_message() == "Initial commit");
+    }
+
+
+    /**
+     * Check the general stag function "add".
+     * 1) Load existing repository (in contrast to initialize in first section)
+     * 2) Create files after repository loading (in contrast to first section)
+     * 3) Check if files appear as untracked
+     * 4) Stage all files. Now the status should show 4 staged new files
+     */
+    SECTION("Stage files")
+    {
+        // Create Git Library
+        GitRepository gl{"sequences"};
+
+        create_testfiles("unit_test_2", 2, "Stage");
+
+        std::vector<FileStatus> stats = gl.status();
+        REQUIRE(stats.size() != 0);
+        for (size_t i =0; i < stats.size(); i++)
+        {
+            FileStatus elm = stats.at(i);
+            if (elm.path_name.rfind("unit_test_1", 0) == 0 || elm.path_name.rfind("unit_test_2", 0) == 0)
```

Comment:
Could maybe be made more readable with `gul14::ends_with()`.

Reply:
I was looking for an endswith function, could not find it. I still cannot find any documenation about gul14::ends_with. Are you sure it exists?